gapidapk/gapidapk.go

```diff
@@ -228,19 +230,22 @@ func ensurePerfettoProducerLaunched(ctx context.Context, d adb.Device) error {
 
 	startSignal, startFunc := task.NewSignal()
 	startFunc = task.Once(startFunc)
+	launchErr := make(chan error)
```

Comment:
You are always sending 2 messages on this channel, but only ever read one. Please clean it up so that it doesn't send it twice. You can also get rid of the start signal, since it's waited on/signaled at the same time as a message is sent on the channel. Having both could lead to a deadlock.